app/parse.py

```diff
@@ -15,8 +26,93 @@ class Product:
     num_of_reviews: int
 
 
+def parse_single_product(soup: BeautifulSoup) -> Product:
+    title = soup.select_one(".title")["title"]
+    description = soup.select_one(".description").text.replace("\xa0", " ")
+    price = soup.select_one(".price").text.replace("$", "")
+    rating = soup.select(".ws-icon")
+    num_of_reviews = soup.select_one(".ratings > .review-count").text
+    return Product(
+        title=title,
+        description=description,
+        price=float(price),
+        rating=len(rating),
+        num_of_reviews=int(num_of_reviews.split()[0])
+    )
+
+
+def get_product_with_pagination_button(url: str) -> [Product]:
```

Comment:
The return type hint '[Product]' is not a valid syntax in Python. Use 'List[Product]' instead and make sure to import 'List' from 'typing'.